### PR TITLE
Improve the settings page

### PR DESCRIPTION
**Pull request by Aller05**
## 1. The plug-in position can be adjusted to any desired position by dragging and dropping.
image
image

## 2. Added the function of customizing the local saving directory address;
image
image
image
image

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -262,13 +237,37 @@ async def get_system_dir(request):
...
+        return web.Response(text=json.dumps({"error": str(e)}), status=500)
+
+def get_my_workflows_dir():
+    data = read_table('userSettings')
+    if(data):
+        records = json.loads(data)
+        curDir = records['myWorkflowsDir'] if records else None
+        if curDir:
+            return curDir
+    return os.path.join(comfy_path,'my_workflows')
```
Comment (Weixuanf) [inline __init__.py:249]: its probably better to keep the const DEFAULT_MY_WORKFLOWS_DIR and use the const here, better readability and easier to change in future

Comment (Aller05) [reply __init__.py:249]: 我改了下这里的逻辑，之前DEFAULT_MY_WORKFLOWS_DIR是有多处引用，所以抽出来比较好。现在默认目录只有这一处引用了，所以删掉了原来的DEFAULT_MY_WORKFLOWS_DIR

File: ui/src/RecentFilesDrawer/WorkspaceSettingsModal.tsx [hunk 2]
```diff
@@ -1,63 +1,271 @@
...
+      setIsEditDirectory(true);
+      setDirPathList(
+        (currentDirectory && currentDirectory.split("/").filter((p) => !!p)) ||
+          []
+      );
+    }
+  };
 
-  const getDir = async (event: any) => {
-    const dir = await getMyWorkflowsDir();
```
Comment (Weixuanf) [inline ui/src/RecentFilesDrawer/WorkspaceSettingsModal.tsx:27]: I feel you need to keep this getMyWorkflowsDir endpoint in Api.ts. i wonder do user always have this value  `userSettingsTable?.records.myWorkflowsDir` in userSettingsTable, even the first time when user just started using workspace-manager and if they have never opened up settings page?

Comment (Aller05) [reply ui/src/RecentFilesDrawer/WorkspaceSettingsModal.tsx:27]: image
新用户首次使用的兜底处理在这里

Comment (Weixuanf) [reply ui/src/RecentFilesDrawer/WorkspaceSettingsModal.tsx:27]: ummm ok这样子倒也work

File: __init__.py (hunk 1 above)
Comment (Weixuanf) [reply __init__.py:249]: 嗯嗯我又理解了一下你这个逻辑，那你这样也make sense，写个const唯一的好处可能就是明显一点儿吧，可读性好点儿，不是必须的

或者你也可以写一个这个function内部的constant FALL_BACK_WORKFLOW_DIR 然后只在这个function内部用

**Pull request merged**